### PR TITLE
New article: CPAN::Audit

### PR DESCRIPTION
**Pull request by briandfoy**
This is as far as I got tonight, but that's about  how it will look.

### CODE REVIEW

**Comment by thibaultduponchelle** [comment]
Preview:
audit

**Comment by thibaultduponchelle** [comment]
The Christmas 🎅🏼 story is missing, but you know this 😄

**Comment by thibaultduponchelle** [comment]
I like CPAN::Audit :heart: 

Verified the links
Verified the code snippets

**Review by thibaultduponchelle — CHANGES_REQUESTED** [review]

**Comment by briandfoy** [comment]
> The Christmas 🎅🏼 story is missing, but I think you know this 😄

I've never made a Christmas story in an advent article. Is that something that has to be there?

**Comment by oalders** [comment]
> I've never made a Christmas story in an advent article. Is that something that has to be there?

Judging by past articles, it seems to come down to the author's preference.

**Comment by oodler577** [comment]
fyi, people might wonder which bdf this is 😄, obviously author's preference

image

**Comment by oodler577** [comment]
> @oodler577 I've missed what you're trying to convey. What do you think needs to change?

Sorry for not being clear. I was wondering if the default _gravitar_ was what was wanted, that's all.

image

File: 2022/incoming/cpan-audit.pod [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+created a way to collate information from MITRE (the repository of CVEs),
+namespace mappings from PAUSE, and version data from MetaCPAN.
+
+The advisory data are actually kept in a separate repository,
+L<cpan-security-advisory|https://github.com/briandfoy/cpan-security-advisory>.
+Each advisory references some vulnerability, notes the affected version,
+and tracks some other information. These are kept in YAML files for
+each distribution.
+
+L<CPAN::Audit> digests those YAML files to create L<CPAN::Audit::DB>
```
[the remark this answers is not in the document]
Comment (briandfoy) [reply 2022/incoming/cpan-audit.pod]: There should all be fixed to just `C<>` now.

**Comment by briandfoy** [comment]
The gravatar issue should be fixed.

**Comment by briandfoy** [comment]
I'm pretty close to the final edits on this article, and I'll read it again tomorrow. I'm not very precious about the edits, so if someone wants to adjust and publish it, go for it. Until then I might fiddle with it a bit more.

File: 2022/incoming/cpan-audit.pod [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+How do you know that you are up-to-date on security advisories for
+perl and CPAN modules? L<CPAN::Audit> and its B<cpan-audit> program
+can do that for you:
+
+    % cpan CPAN::Audit
+	% cpan-audit installed
+
+You end up with a list of reports of known vulnerabilities for
+everything in your `@INC`. Fix some of those by upgrading modules,
```
Comment (thibaultduponchelle) [inline 2022/incoming/cpan-audit.pod:15]:
```suggestion
everything in your C<@INC>. Fix some of those by upgrading modules,
```

File: 2022/incoming/cpan-audit.pod [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+using a more modern replacement, or, even better fix the problem and
+send that upstream.
+
+These reports come mostly from L<MITRE|https://www.cve.org> and it's
+library of Common Vulnerabilities and Exposures, or just CVE, that's
+it's developed since 1999. All of this is pre-digested and arranged
+into C<CPAN::Audit::DB>, which comes with the L<CPAN::Audit>
+distribution.
+
+=head Check a module
```
Comment (thibaultduponchelle) [inline 2022/incoming/cpan-audit.pod:25]:
```suggestion
=head2 Check a module
```

**Review by thibaultduponchelle — CHANGES_REQUESTED** [review]

File: 2022/incoming/cpan-audit.pod (hunk 3 above)
Comment (thibaultduponchelle) [reply 2022/incoming/cpan-audit.pod:25]: This is what is reported by failed GitHub action

**Comment by thibaultduponchelle** [comment]
I did some suggestions, overall sounds good to me :+1:

File: 2022/incoming/cpan-audit.pod [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+Topic: CPAN::Audit
+
+=encoding utf8
+
+How do you know that you are up-to-date on security advisories for
+perl and CPAN modules? L<CPAN::Audit> and its B<cpan-audit> program
+can do that for you:
+
+    % cpan CPAN::Audit
+	% cpan-audit installed
```
Comment (oalders) [inline 2022/incoming/cpan-audit.pod:12]:
```suggestion
    % cpan CPAN::Audit
    % cpan-audit installed
```

File: 2022/incoming/cpan-audit.pod [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+
+You end up with a list of reports of known vulnerabilities for
+everything in your C<@INC>. Fix some of those by upgrading modules,
+using a more modern replacement, or, even better fix the problem and
+send that upstream.
+
+These reports come mostly from L<MITRE|https://www.cve.org> and it's
```
Comment (oalders) [inline 2022/incoming/cpan-audit.pod:19]:
```suggestion
These reports come mostly from L<MITRE|https://www.cve.org> and its
```

File: 2022/incoming/cpan-audit.pod [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+library of Common Vulnerabilities and Exposures, or just CVE, that's
```
Comment (oalders) [inline 2022/incoming/cpan-audit.pod:20]:
```suggestion
library of Common Vulnerabilities and Exposures, or just CVE, that
```

**Review by oalders — APPROVED** [review]

**Comment by oalders** [comment]
Thanks @briandfoy and @thibaultduponchelle!

**Pull request merged**